src/modules/Dashboard/features/Applet/Popups/AddParticipantPopup/PublicLinkPopup/PublicLinkPopup.tsx

```diff
@@ -0,0 +1,113 @@
+import { Button } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+
+import { deleteAppletPublicLinkApi, postAppletPublicLinkApi } from 'api';
+import { Modal, Spinner } from 'shared/components';
+import { useAsync } from 'shared/hooks';
+import { StyledBodyLarge, StyledFlexAllCenter, StyledFlexColumn } from 'shared/styles';
+
+import { PublicLinkPopupProps } from './PublicLinkPopup.types';
+
+export const PublicLinkPopup = ({
+  appletId,
+  'data-testid': dataTestId,
+  hasPublicLink = false,
+  onClose,
+  ...otherProps
+}: PublicLinkPopupProps) => {
+  const { t } = useTranslation('app');
+  const {
+    execute: createPublicLink,
+    isLoading: createIsLoading,
+    error: createError,
+  } = useAsync(postAppletPublicLinkApi, (res) => {
+    if (res?.data?.result) {
+      onClose?.(true);
+    }
+  });
+  const {
+    execute: deletePublicLink,
+    isLoading: deleteIsLoading,
+    error: deleteError,
+  } = useAsync(deleteAppletPublicLinkApi, (res) => {
+    if (res?.status === 204) {
+      onClose?.();
+    }
+  });
+  const isLoading = createIsLoading || deleteIsLoading;
+  const hasError = (createError || deleteError) && !isLoading;
+
+  const handleCreatePublicLink = (requireLogin: boolean = false) => {
+    if (appletId) {
+      createPublicLink({ appletId, requireLogin });
+    }
+  };
+
+  const handleDeletePublicLink = () => {
+    if (appletId) {
+      deletePublicLink({ appletId });
+    }
+  };
+
+  return (
+    <Modal
+      data-testid={dataTestId}
+      footer={
+        <StyledFlexAllCenter sx={{ gap: 1.6, width: '100%' }}>
+          {hasPublicLink ? (
+            <Button
+              color="error"
+              data-testId={dataTestId && `${dataTestId}-delete-bttn`}
```

Comment:
Nit: I've only ever seen buttons abbreviated as `btn` in code (despite their two "t"s). Seems `btn` is also the de-facto abbreviation used in the codebase. NBD, just a stickler for consistency. 😛

Reply:
Hah — every instance of `bttn` was added by me. Aside from the changes in this PR, there's some variables in a test file that I named with the two-t style a few weeks ago. I went ahead and [renamed all of them](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1727/commits/6b7148f41bd4370dbce091f3bb29fc28171c4ae2).